### PR TITLE
miscellaneous fixes to the OVN logical topology upgrade logic

### PR DESCRIPTION
**Pull request by girishmg**
- fix typo in OvnSingleJoinSwithTopoVersion constant
- for cleanup look only for `join_<nodeName>` logical swtiches
- don't depend on `error message` has an API since they might change
  underneath us

@trozet @cathy-zhou PTAL

### CODE REVIEW

File: go-controller/pkg/ovn/master.go [hunk 1]
```diff
@@ -176,28 +174,38 @@ func (oc *Controller) upgradeToSingleSwitchOVNTopology(existingNodeList *kapi.No
...
-			// no OVNClusterRouter exists, DB is empty, nothing to upgrade
-			ver = OvnCurrentTopologyVersion
-		} else {
-			klog.Infof("No version string found. The OVN topology is before versioning is introduced. Upgrade needed")
-		}
+		return fmt.Errorf("failed in retrieving %s to determine the current version of OVN logical topology: "+
+			"stderr: %q, error: %v", util.OVNClusterRouter, stderr, err)
+	}
+	if len(stdout) == 0 {
+		// no OVNClusterRouter exists, DB is empty, nothing to upgrade
```
Comment (cathy-zhou) [inline go-controller/pkg/ovn/master.go:187]: Then ver shoud be set to OvnCurrentTopologyVersion in this case, right?

Comment (girishmg) [reply go-controller/pkg/ovn/master.go:187]: We return from here and then when the ovn_cluster_router is created later on, there we will set the external_ids:k8s-ovn-topo-version to current version.

Comment (cathy-zhou) [reply go-controller/pkg/ovn/master.go:187]: ok

**Comment by cathy-zhou** [comment]
lgtm

**Pull request merged**